tests/rbd_mirror/test_rbd_mirror_snapshot_metrics.py

```diff
@@ -0,0 +1,158 @@
+
+        # Check if the symlink file named rbd-mirror.asok is in the directory_contents
+        if "rbd-mirror.asok" in directory_contents:
+            log.info("symlink file named rbd-mirror.asok created successfully")
+        else:
+            log.error("symlink file named rbd-mirror.asok not created successfully")
+            return 1
+
+    # Validation of mirror related metrics
+    command = "ceph --admin-daemon rbd-mirror.asok counter dump"
+
+    out = mirror_node.exec_cmd(cmd=command, output=True)
+
+    # Check if the required snapshot based metrics are present
+    if (
+        "rbd_mirror_snapshot" in out
+        and "labels" in out["rbd_mirror_snapshot"]
+        and "counters" in out["rbd_mirror_snapshot"]
+    ):
+        counters = out["rbd_mirror_snapshot"]["counters"]
+        if (
+            "snapshots" in counters
+            and "sync_time" in counters
+            and "sync_bytes" in counters
+        ):
+            log.info("All required snapshot based metrics are present.")
+            return 0
+    else:
+        log.error("One or more required snapshot based metrics are missing.")
+        return 1
+
+
+def run(ceph_cluster, **kw):
+    """Verification of mirroring failure,
+    when secondary cluster storage is full.
+
+    Args:
+        ceph_cluster: The storage cluster participating in the test
+        **kw: test data
+
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+
+    try:
+        log.info(
+            "Starting CEPH-9507 - Verification of mirroring failure "
+            "when secondary cluster storage is full"
+        )
+        rbd = Rbd(**kw)
+        config = kw.get("config")
+        mode = config.get("mode")
+        mirrormode = config.get("mirrormode")
+        poolname = rbd.random_string() + "_test_pool"
+        imagename = rbd.random_string() + "_test_image"
+
+        rbd1, rbd2 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw.get("ceph_cluster_dict").keys()
+        ]
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd1"), config
+        )
+        mirror2 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd2"), config
+        )
+
+        mirror1.initial_mirror_config(
+            mirror2,
+            poolname=poolname,
+            imagename=imagename,
+            imagesize=config.get("imagesize", "10G"),
+            io_total=config.get("io-total", "4G"),
+            mode=mode,
+            mirrormode=mirrormode,
+            peer_mode=config.get("peer_mode", "bootstrap"),
+            rbd_client=config.get("rbd_client", "client.admin"),
+            build=config.get("build"),
+            **kw,
+        )
+
+        mirror1.mirror_snapshot_schedule_add(poolname=poolname, interval="3m")
+
+        # To get rbd-mirror daemon node
+        mirror_node = [ceph_cluster.get_nodes(role="rbd-mirror")[0]]
```

Comment:
Do you want mirror node to be a list or just one value?

Reply:
Modified.